aws_lambda_powertools/utilities/parser/envelopes/sns.py

```diff
@@ -69,6 +69,6 @@ def parse(self, data: Optional[Union[Dict[str, Any], Any]], model: Type[Model])
         parsed_envelope = SqsModel.parse_obj(data)
         output = []
         for record in parsed_envelope.Records:
-            sns_notification = SnsNotificationModel.parse_raw(record.body)
+            sns_notification: SnsNotificationModel = SnsNotificationModel.parse_raw(cast(str, record.body))
```

Comment:
can we solve without casting?

Reply:
Sure, I can explicitly check if body is of type str and do this parsing then and fail if its of type `Type[BaseModel]`

Reply:
Yeah I see why! Let's add a temporary variable with casting. I was wrong about `cast` as it's [only returning the value](https://github.com/python/cpython/blob/main/Lib/typing.py#L1712) now not like `str()`. 

```python
            # because we allow either AWS expected contract (str) or a custom Model, see #943
            d = cast(str, record.body)
            sns_notification = SnsNotificationModel.parse_raw(d)
```

Reply:
ok, lets create temp variable. Thx!